src/esp/assets/ResourceManager.cpp

```diff
@@ -1197,12 +1166,86 @@ void ResourceManager::loadMaterials(Importer& importer,
       continue;
     }
 
-    // using make_unique will not work here
-    std::unique_ptr<Magnum::Trade::PhongMaterialData> phongMaterialData(
-        static_cast<Magnum::Trade::PhongMaterialData*>(materialData.release()));
+    const auto& phongMaterialData =
+        static_cast<Mn::Trade::PhongMaterialData&>(*materialData);
+    std::unique_ptr<gfx::MaterialData> finalMaterial;
+    if (forceFlatShading) {
+      finalMaterial = getFlatShadedMaterialData(phongMaterialData,
+                                                metaData->textureIndex.first);
+    } else {
+      finalMaterial = getPhongShadedMaterialData(phongMaterialData,
+                                                 metaData->textureIndex.first);
+    }
+    // for now, just use unique ID for material key. This may change if we
+    // expose materials to user for post-load modification
+    shaderManager_.set(std::to_string(currentMaterialID),
+                       finalMaterial.release());
+  }
+}
+
+gfx::PhongMaterialData::uptr ResourceManager::getFlatShadedMaterialData(
+    const Mn::Trade::PhongMaterialData& material,
+    int textureBaseIndex) {
+  // NOLINTNEXTLINE(google-build-using-namespace)
+  using namespace Mn::Math::Literals;
+
+  auto finalMaterial = gfx::PhongMaterialData::create_unique();
+  finalMaterial->ambientColor = 0xffffffff_rgbaf;
+  finalMaterial->diffuseColor = 0x00000000_rgbaf;
+  finalMaterial->specularColor = 0x00000000_rgbaf;
+
+  if (material.flags() & Mn::Trade::PhongMaterialData::Flag::AmbientTexture) {
+    finalMaterial->ambientTexture =
+        textures_[textureBaseIndex + material.ambientTexture()].get();
+  } else if (material.flags() &
+             Mn::Trade::PhongMaterialData::Flag::DiffuseTexture) {
+    // if we want to force flat shading, but we don't have ambient texture,
+    // check for diffuse texture and use that instead
+    finalMaterial->ambientTexture =
+        textures_[textureBaseIndex + material.diffuseTexture()].get();
+  } else {
+    finalMaterial->ambientColor = material.ambientColor();
+  }
+  return finalMaterial;
+}
+
+gfx::PhongMaterialData::uptr ResourceManager::getPhongShadedMaterialData(
+    const Mn::Trade::PhongMaterialData& material,
+    int textureBaseIndex) {
+  // NOLINTNEXTLINE(google-build-using-namespace)
+  using namespace Mn::Math::Literals;
+
+  auto finalMaterial = gfx::PhongMaterialData::create_unique();
+  // TODO: figure out why materials are being loaded with shininess == 1
```

Comment:
Which file format? I know Assimp was doing some shady business for certain formats :)

Reply:
actually, every single asset I've tried to load has shininess == 1, ie. `data/objects/banana.glb`, `data/objects/cheezit.glb`

Reply:
Can you try the other test objects? All the test objects we provide are converted obj -> glb YCB objects.

Reply:
It's actually done by Magnum's TinyGltfImporter sorry. It attempts to translate the PBR materials to a basic Phong shading and leaves shininess at 1 unless the model defines also a Blinn/Phong shading model.

I'll patch this to a better default and remove this TODO once we upgrade to latest magnum.

Reply:
Thanks for identifying the cause, Mosra. 

In the meantime it makes sense to provide an easy handle for manipulating the material so users can edit this (follow-up PR?). 

It may also be better to assume low/no shininess for most objects (furniture/cardboard/etc). Let's iterate on the default based on a set of test objects.

Reply:
The default is fixed in #496 :)